client/gutenberg/editor/components/header/header-toolbar/index.js

```diff
@@ -50,8 +51,17 @@ function HeaderToolbar( {
 	// GUTENLYPSO END
 } ) {
 	const onCloseButtonClick = () => {
-		notices.forEach( ( { id } ) => removeNotice( id ) );
-		closeEditor();
+		// GUTENLYPSO START
+		const proceed =
+			isDirty && typeof window !== 'undefined'
+				? window.confirm( __( 'Changes that you made may not be saved.' ) )
```

Comment:
Does this get translated?

Reply:
I suppose it does since we use the same call throughout Gutenlypso codebase. If it's not we'll have to change all translations in a separate PR.